File: bfs_dfs_problems/bfs_dfs_problems__move_once.cs
// 実行時間 20ms
using static System.Console;
using static System.Linq.Enumerable;

internal class Program
{
    static void Main()
    {
        int[] hw = ReadLine().Split().Select( int.Parse ).ToArray();
        int[] yx = ReadLine().Split().Select( int.Parse ).ToArray();
        int[] dx = { 0, 0, -1, 1 };
        int[] dy = { -1, 1, 0, 0 };
        char[][] g = new char[hw[0]][];

        foreach( int i in Range( 0, hw[0] ) ) {
            g[i] = Repeat( '.', hw[1] ).ToArray();
        }
        g[yx[0]][yx[1]] = '*';
        foreach( int i in Range( 0, 4 ) ) {
            int x = yx[1] + dx[i];
            int y = yx[0] + dy[i];

            if( 0 <= y && y < hw[0] && 0 <= x && x < hw[1] ) {
                g[y][x] = '*';
            }
        }
        foreach( char[] c in g ) {
            WriteLine( string.Join( "", c ) );
        }
        return;
    }
}

File: bfs_dfs_problems/bfs_dfs_problems__move_once.py
# 実行時間 10ms
h, w = list(map(int, input().split()))
y, x = list(map(int, input().split()))
dx = [0, 0, -1, 1]
dy = [-1, 1, 0, 0]
g = [["." for _ in [0] * w] for _ in [0] * h]

g[y][x] = "*"
for i in range(4):
    a = x + dx[i]
    b = y + dy[i]

    if 0 <= b < h and 0 <= a < w:
        g[b][a] = "*"
for c in g:
    print("".join(c))
